astropy/config/paths.py

```diff
@@ -195,18 +227,31 @@ def _find_or_create_root_dir(
             raise OSError(
                 f"Intended {pkgname} {cls._directory_type} directory {maindir} is actually a file."
             )
+
+        warn = partial(
+            cls._warn_env_var_is_ignored, env_var_name=cls._directory_env_var
+        )
+        if linkto is not None:
+            if sys.platform.startswith("win"):
+                warn("is not supported on Windows")
+                linkto = None
+            elif maindir.is_dir():
+                warn(
+                    f"the default location, {maindir}, already exists, and takes precedence"
+                )
```

Comment:
The more I think about it, the more I think we do not need warning here. The test would need to be adjusted if we remove this warning.
```suggestion
```